### PR TITLE
[pyupgrade] `groups/`, `federation/`, `events/`, and `crypto/`

### PR DESCRIPTION
**Pull request by ShadowJonathan**
Part of #9744

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

`Signed-off-by: Jonathan de Jong <jonathan@automatia.nl>`

### CODE REVIEW

File: changelog.d/9749.misc [hunk 1]
```diff
@@ -0,0 +1 @@
+Run `pyupgrade --py36-plus` over the codebase.
```
Comment (ShadowJonathan) [inline changelog.d/9749.misc:1]: This'll be the newsfile that i will use across all `pyupgrade` PRs, any adjustments?

**Comment by ShadowJonathan** [comment]
I'm opening this one - the smallest PR out of the bunch - first, so that any misunderstandings and gotchas can be caught on this one first, before they'd happen across other PRs.

**Comment by clokep** [comment]
Just to reconfirm, this is equivalent to running `pyupgrade --py36-plus synapse/groups synapse/federation synapse/events synapse/crypto`? I.e. all changes are automated from pyupgrade?

**Comment by ShadowJonathan** [comment]
Yes, though it's more akin to running `find $1 -type f -name "*.py"` on each, as the command doesn't recurse into directories (many PRs requesting such feature nonwithstanding)

I also ran `scripts-dev/lint.sh` after it, and some complaints from that weren't automatically fixed (# of empty lines before class declarations with comments) on some of the other PRs, but yes, every change is automated.

**Comment by ShadowJonathan** [comment]
Closed in favour of better refactoring methodology, see #9744

**Pull request closed without merging**